### DISCUSSION TITLE
1BRC—The Results Are In! - Gunnar Morling

### DISCUSSION THREAD

**Discussion by giscus** [gunnarmorling/discussions.morling.dev#314] [Announcements]
# 1BRC—The Results Are In! - Gunnar Morling

Oh what a wild ride the last few weeks have been.
The One Billion Row Challenge (1BRC for short),
something I had expected to be interesting to a dozen folks or so at best,
has gone kinda viral, with hundreds of people competing and engaging.
In Java, as intended, but also beyond:
folks implemented the challenge in languages such as Go, Rust, C/C++, C#, Fortran, or Erlang, as well databases (Postgres, Oracle, Snowflake, etc.), and tools like awk.

It’s really incredible how far people have pushed the limits here.
Pull request by pull request, the execution times for solving the problem layed out in the challenge — aggregating random temperature values from a file with 1,000,000,000 rows — improved by two orders of magnitudes in comparison to the initial baseline implementation.
Today I am happy to share the final results, as the challenge closed for new entries after exactly one month on Jan 31
and all submissions have been reviewed.

https://www.morling.dev/blog/1brc-results-are-in/

**Comment by armandino** [comment]
Thank you, Gunnar, for organising such a fun challenge. I have 1brc withdrawal symptoms now :sweat_smile:

**Reply by gunnarmorling** [reply]
Hey @armandino, thanks a lot for your nice words and being a part of 1BRC! I how your withdrawal symptoms will be over soon ;)

**Comment by thenextsupersun** [comment]
Thank you, Gunnar, for organising such a fun challenge. There is another puzzle confused me a lot ,
Arrays, linked lists, and red-black trees each perform insert, search, and delete operations on 10 million data items. Which one consumes the least time in the end?

**Comment by tianzhipeng-git** [comment]
A very interesting challenge, unfortunately I saw it too late, but I still have to leave a 👍🏻
